device/esp_tinyusb/Kconfig

```
@@ -285,6 +285,41 @@ menu "TinyUSB Stack"
             config TINYUSB_NET_MODE_NONE
                 bool "None"
         endchoice
+
+        config TINYUSB_NCM_OUT_NTB_BUFFS_COUNT
+            int "Number of NCM NTB buffers for reception side"
+            depends on TINYUSB_NET_MODE_NCM
+            default 2
+            range 1 4
+            help
+                Number of NTB buffers for reception side.
+                Can be increased to improve performance with the cost of additional RAM requirements.
+
+        config TINYUSB_NCM_IN_NTB_BUFFS_COUNT
+            int "Number of NCM NTB buffers for transmission side"
+            depends on TINYUSB_NET_MODE_NCM
+            default 2
```

IMHO it is better to use the best default value, which is 3.

https://github.com/espressif/tinyusb/blob/37995c70595164aa82f5b13496ae71c9584224b9/src/class/net/ncm.h#L63

```suggestion
            default 3
```

---

The suggestions from tinyusb are based on `RPi RP2040`.  And for the [STM32](https://github.com/espressif/tinyusb/blob/37995c70595164aa82f5b13496ae71c9584224b9/src/class/net/ncm.h#L66) for example, there is no performance gain at all.

I did some Iperf testing, but there was no such performance gain in 50%, or 30%, as the description in tinyusb for RP2040 is claiming.

Also the statement [3 - "tud_network_can_xmit: request blocked" never happens](https://github.com/espressif/tinyusb/blob/37995c70595164aa82f5b13496ae71c9584224b9/src/class/net/ncm.h#L63) is not true, in our case. I had to set the buffer count really high (to around 10) to completely mitigate the warning message.

Sure, we can have the default value 3, but for esp32, we can't really say that there is some visible performance gain, or `request blocked` will never happen.

---

Oh, cool.
Then, if we have the real test results, why we can't use them? 
That means, that for our case, to exclude the blocking, we need configure it to 10. 

BTW, is that the same for Slave and DMA mode?

---

Bit difficult to say what is the optimal count here.
 
Wen running IPERF, setting the NTB Count to 2 or 3 was enough, to get the NCM stable (no `request blocked` happened at all).

But then, I used esp32s3 as a network device [sta2eth](https://github.com/espressif/esp-idf/tree/master/examples/network/sta2eth) example or [tusb_ncm](https://github.com/espressif/esp-idf/tree/master/examples/peripherals/usb/device/tusb_ncm) example. So practically I used esp32s3 connected to a local wifi network, forwarding connection to my PC over USB for casual Internet browsing (google maps, youtube... ) In this case, I had to set the buff count to 10, to stop seeing `request blocked`. And this looks like a bit of extreme scenario to me.

The DMA settings had no effect on this if I remember. I can check again.